### PR TITLE
feat: PHP 8.2 support

### PR DESCRIPTION
**Pull request by falvarez**
Solves #54 .

I inferred the changes looking into https://github.com/php/php-src/pull/8327.

### CODE REVIEW

**Comment by dmitryuk** [comment]
What about to add php 8.1 and 8.2 to .travis.yml ?

**Comment by falvarez** [comment]
> What about to add php 8.1 and 8.2 to .travis.yml ?

Thanks for the suggestion.

I blindly added both versions to `.travis.yml`, as I have no experience using it.

Hope it will work.

**Comment by angeljqv** [comment]
Any news about this??

**Comment by angeljqv** [comment]
Also change to `<max>8.2.99</max> `
https://github.com/php-zookeeper/php-zookeeper/blob/249f790c22700d684d6a7b70c5d873ea0d82ca6b/package.xml#L79-L82

**Comment by falvarez** [comment]
> Also change to `<max>8.2.99</max> `
> 
> https://github.com/php-zookeeper/php-zookeeper/blob/249f790c22700d684d6a7b70c5d873ea0d82ca6b/package.xml#L79-L82

Done. Thank you for your suggestion.

File: php_zookeeper.c [hunk 1]
```diff
@@ -1012,10 +1012,14 @@ void php_zk_watcher_marshal(zhandle_t *zk, int type, int state, const char *path
 #if PHP_VERSION_ID >= 70100
 #ifdef ZTS
 	(*cb_data->vm_interrupt) = 1;
+#else
+#if PHP_VERSION_ID >= 80200
+  EG(vm_interrupt) = (zend_atomic_bool){.value=1};
```
Comment (Timandes) [inline php_zookeeper.c:1017]: Is there any possibility that this should be:
```c
zend_atomic_bool_store_ex(&EG(timed_out), true);
```

:)

Comment (angeljqv) [reply php_zookeeper.c:1017]: @falvarez ping

Comment (angeljqv) [reply php_zookeeper.c:1017]: @falvarez ping

**Comment by zhupeijie** [comment]
您好，邮件已收到，我近期会登录查看，祝您天天开心！

File: php_zookeeper.c (hunk 1 above)
Comment (falvarez) [reply php_zookeeper.c:1017]: Do you mean?

```diff
- EG(vm_interrupt) = (zend_atomic_bool){.value=1};
+ zend_atomic_bool_store_ex(&EG(timed_out), true); 
```

Well, I don't really know. As I said, I have no idea of what I'm doing. I've done blind changes trying to compile the extension for PHP 8.2. 🤷‍♂️

Comment (angeljqv) [reply php_zookeeper.c:1017]: Did you compile it and did it work on PHP 8.2?

**Pull request merged**